server/src/main/java/org/candlepin/util/X509Util.java

```diff
@@ -183,41 +172,8 @@ public String createFullContentPath(String contentPrefix, ProductContent pc) {
         }
 
         for (ProductContent pc : pcSet) {
-            boolean canUse = true;
-            Set<String> contentArches = Arch.parseArches(pc.getContent().getArches());
-            Set<String> productArches =
-                Arch.parseArches(product.getAttributeValue(Product.Attributes.ARCHITECTURE));
-
-            // Empty or null Content.arches should result in
-            // inheriting the arches from the product
-            if (contentArches.isEmpty()) {
-                // No content arch, see if there is a Product arch
-                // and if so inherit it.
-                if (!productArches.isEmpty()) {
-                    contentArches.addAll(productArches);
-                }
-                else {
-                    // No Product arches either, log it, but do
-                    // not filter out this content
-                    log.debug("No arch attributes found for content or product");
-                }
-            }
-
-            searchArch: {
-                for (String contentArch : contentArches) {
-
-                    for (String consumerArch : consumerArches) {
-
-                        if (Arch.contentForConsumer(contentArch, consumerArch)) {
-                            canUse = true;
-                            break searchArch;
-                        }
-                        else {
-                            canUse = false;
-                        }
-                    }
-                }
-            }
+            Set<String> contentArches = archesOf(product, pc);
+            boolean canUse = hasCompatibleArchitecture(consumerArches, contentArches);
```

Comment:
Good job on refactoring this method for readability, but it seems there were no unit tests for it. Let's add some unit tests now, since refactoring code without existing unit tests is prime for regressions appearing

Reply:
Done.